controllers/toolchainconfig/deliveryserviceconfig.go

```diff
@@ -0,0 +1,25 @@
+package toolchainconfig
+
+type DeliveryServiceFactoryConfig struct {
+	ToolchainConfig
+}
+
+func (d DeliveryServiceFactoryConfig) GetNotificationDeliveryService() string {
+	return d.Notifications().NotificationDeliveryService()
+}
+
+func (d DeliveryServiceFactoryConfig) GetMailgunDomain() string {
+	return d.Notifications().MailgunDomain()
+}
+
+func (d DeliveryServiceFactoryConfig) GetMailgunAPIKey() string {
+	return d.Notifications().MailgunAPIKey()
+}
+
+func (d DeliveryServiceFactoryConfig) GetMailgunSenderEmail() string {
+	return d.Notifications().MailgunSenderEmail()
+}
+
+func (d DeliveryServiceFactoryConfig) GetMailgunReplyToEmail() string {
+	return d.Notifications().MailgunReplyToEmail()
+}
```

Comment:
Do we still need this level of abstraction? It looks overcomplicated to me and setting up the tests should be pretty easy now.

In addition, in this way we won't be able to modify any of the values on the fly - we would need to restart the host-operator deployment after doing some change.

Reply:
I agree with you but I think it was done intentionally this way so that the notification service can swap out one mail provider with another without requiring many changes? Maybe we should discuss it in a sync call and do that work separately if we want to remove the abstraction.

Reply:
As was discussed - keep it as it is in this PR and let's try to remove it in separate one.